### PR TITLE
feat(cicd): add pr title validation

### LINKED ISSUE #38: [CICD] - add PR title validation
### Start Date

_No response_

### Implementation PR

NA

### Reference Issues

N/A

### Summary

- Create `.github/workflows/pr-title-checker.yaml`
- Trigger the validation when:
  - opened
  - reopened
  - edited
  - synchronize

### Basic Example

```yaml
steps:
  - uses: aslafy-z/conventional-pr-title-action@v2
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

### Drawbacks

N/A

### Unresolved questions

_No response_

### PR DESCRIPTION
**Pull request by tiagomichaelsousa**
## Description

Add PR title validation to enforce conventional commits

## Related Issue

Closes #38

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/pr-title-checker.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  title-checker:
+    name: Check PR title
+    runs-on: [ self-hosted ]
+    
+    steps:
+      - uses: aslafy-z/conventional-pr-title-action@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (comoser) [inline .github/workflows/pr-title-checker.yaml:23]: Do I need to create something regarding this `GITHUB_TOKEN` or have you already done this?

Comment (tiagomichaelsousa) [reply .github/workflows/pr-title-checker.yaml:23]: You don't have do to anything, this is one of the default GitHub env vars 😎

Comment (comoser) [reply .github/workflows/pr-title-checker.yaml:23]: perfect! thank you for this :D

**Review by comoser — APPROVED** [review]
LGTM 🎉

**Pull request merged**